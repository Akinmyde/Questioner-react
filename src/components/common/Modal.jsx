import React from 'react';

const Modal = ({ message, onDelete, onCancel }) => {
  return (
    <div className="modal show-modal">
      <div className="modal-content">
        <span className="close-button" onClick={onCancel}>&times;</span>
        <p className="msg">{message}</p>
        <div>
          <button className="btn-modal btn-yes" onClick={onDelete}>Yes</button>
          <button className="btn-modal btn-no" onClick={onCancel}>No</button>
        </div>
      </div>
    </div>
  );
}
 
export default Modal;