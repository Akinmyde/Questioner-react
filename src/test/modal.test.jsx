import React from 'react';
import { shallow } from 'enzyme';
import Modal from '../components/common/Modal';

describe('HomePage component', () => {
  it('should match snapshot', () => {
    const props = { message: 'test', onDelete: jest.fn(), onCancel: jest.fn() }
    const wrapper = shallow(<Modal {...props} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div'))
  });
});
