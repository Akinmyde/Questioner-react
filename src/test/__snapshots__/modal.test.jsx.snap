// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component should match snapshot 1`] = `
<div
  className="modal show-modal"
>
  <div
    className="modal-content"
  >
    <span
      className="close-button"
      onClick={[MockFunction]}
    >
      Ã—
    </span>
    <p
      className="msg"
    >
      test
    </p>
    <div>
      <button
        className="btn-modal btn-yes"
        onClick={[MockFunction]}
      >
        Yes
      </button>
      <button
        className="btn-modal btn-no"
        onClick={[MockFunction]}
      >
        No
      </button>
    </div>
  </div>
</div>
`;
